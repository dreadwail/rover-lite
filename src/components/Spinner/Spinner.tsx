import React, { FC } from 'react';

export interface SpinnerProps {
  readonly isLargeSize?: boolean;
}

const Spinner: FC<SpinnerProps> = ({ isLargeSize }) => {
  const style = isLargeSize ? { width: '5rem', height: '5rem' } : {};
  return (
    <div className="d-flex justify-content-center m-5">
      <div className="spinner-border text-primary" style={style} role="status">
        <span className="sr-only">Loading...</span>
      </div>
    </div>
  );
};

export default Spinner;
