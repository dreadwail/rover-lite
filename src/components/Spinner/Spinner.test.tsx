import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Spinner from './Spinner';

describe('Spinner', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    rendered = shallow(<Spinner />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });

  describe('when large sized', () => {
    beforeEach(() => {
      rendered = shallow(<Spinner isLargeSize={true} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });
  });
});
