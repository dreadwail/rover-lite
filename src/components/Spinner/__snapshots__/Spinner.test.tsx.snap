// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
<div
  className="d-flex justify-content-center m-5"
>
  <div
    className="spinner-border text-primary"
    role="status"
    style={Object {}}
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`Spinner when large sized renders correctly 1`] = `
<div
  className="d-flex justify-content-center m-5"
>
  <div
    className="spinner-border text-primary"
    role="status"
    style={
      Object {
        "height": "5rem",
        "width": "5rem",
      }
    }
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
